src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -782,6 +825,8 @@ void Simulator::poll_for_MAVLink_messages()
 	_actuator_outputs_sub = orb_subscribe_multi(ORB_ID(actuator_outputs), 0);
 	_vehicle_status_sub = orb_subscribe(ORB_ID(vehicle_status));
 
+	_tune_control_sub = orb_subscribe(ORB_ID(tune_control));
```

Comment:
This and the other topics are missing an `orb_unsubscribe` at the bottom of the method.

Reply:
Hi @bkueng , I think I can see why the CI tests [here](https://github.com/PX4/Firmware/pull/11631/commits) are unhappy... do you have a preference about where the unsubscribe methods get placed?

Reply:
In that case at the end of `poll_for_MAVLink_messages`.

Reply:
Apologies, I didn't make things very clear by pointing to those commits,
> This and the other topics are missing an orb_unsubscribe at the bottom of the method

> In that case at the end of poll_for_MAVLink_messages.

Indeed, they are the same.

Adding the orb unsubscribe calls at the end of this method causes a segfualt.  Placing the unsubscribe() calls in the destructor does not.  Additional usage of the subscription vars occurs in `send()`, `send_controls`, and `poll_topics()`.  The destructor seems to be the appropriate location for the unsubscribe() calls given their usage outside of `poll_for_MAVLINK_messages()`, but I can find no other examples where subscriptions take place outside of the constructor, (in a class method), and then unsubscribed in the destructor.

It seems appropriate to move these subscribe calls to the constructor in this case, but let me know if you have other feedback in PR #11631. Thanks!

Reply:
> Adding the orb unsubscribe calls at the end of this method causes a segfualt. Placing the unsubscribe() calls in the destructor does not.

If I saw it correctly, you called unsubscribe still within the loop, not at the very end of `poll_for_MAVLINK_messages`.

>  The destructor seems to be the appropriate location for the unsubscribe()

If you subscribe in the constructor it's the right place. That's my prefered solution as well, and it should work with the current structure (refactoring to ModuleBase is another thing we can do).